regnety/main.py

```diff
@@ -0,0 +1,42 @@
+import tensorflow as tf
+import os
+import random
+import json
+import argparse
+import math
+from utils.tfrecords_utils import *
+from dataset.imagenet import ImageNet
+import matplotlib.pyplot as plt
+
+def main():
+    parser = argparse.ArgumentParser(description="Make TFRecords")
+    parser.add_argument("--odir", type=str)
+    parser.add_argument("--data_dir", type=str)
+    parser.add_argument("--file_prefix", type=str)
+    parser.add_argument("--synset_filepath", type=str)
+    parser.add_argument("--batch_size", type=int, default=1024)
+
+    args = parser.parse_args()
+    make_tfrecs(
+        dataset_base_dir=args.data_dir,
+        output_dir=args.odir,
+        file_prefix=args.file_prefix,
+        synset_filepath=args.synset_filepath,
+        batch_size = args.batch_size,
+    )
+    # for i in ds:
+    #     print(i)
+    #     break
+        
+
+    imgnet = ImageNet([os.path.join('/content',i) for i in os.listdir('/content') if i.startswith('trial5')])
```

Comment:
I'm a bit confused about the point of this script. If I understand correctly:

`make_tfrecs` reads the ImageNet data, preprocesses it to a consistent format and image size, and saves that as TFRecords. That seems to be all that's needed for one-time data preprocessing.

Then this code runs to augment the images. Why is that happening at the same time? As discussed in our last meeting, it doesn't make sense to perform augmentations such as random cropping once during pre-processing. These augmentations should happen continuously during training.

Reply:
To elaborate a little more, performing random augmentation now just goes from image set A with size `n` to slightly different image set B also with size `n`. This has minimal benefit. Conversely, performing random augmentations continuously during training means that in each epoch you go from image set A of size `n` to some new image set B with size `n`, and then on the next epoch you get new image set C, then D, etc. This effectively multiplies the number and variety of training examples you have.

Reply:
`main.py` is a script that I just use to test all the functions quickly and as a pipeline. Yes, only the preprocessing part (i.e. converting raw data to TFRecords) will be done beforehand. The `ImageNet` class is for training time.